src/ContentAwareGenerator.php

```diff
@@ -68,7 +68,9 @@ public function setContentRepository(ContentRepositoryInterface $contentReposito
      */
     public function generate($name, $parameters = [], $absolute = UrlGeneratorInterface::ABSOLUTE_PATH)
     {
-        if ($name instanceof SymfonyRoute) {
+        if (array_key_exists('_cmf_route', $parameters) && $parameters['_cmf_route'] instanceof SymfonyRoute) {
+            $route = $this->getBestLocaleRoute($parameters['_cmf_route'], $parameters);
+        } elseif ($name instanceof SymfonyRoute) {
```

Comment:
to avoid edge case BC breaks, i would prefer to take the `$name instanceof ...` line as the first condition.

Reply:
Done! I've also switch the condition back in the `ProviderBasedGenerator`

Reply:
i think you switched it below in getRouteByContent, but not here on this line

Reply:
Yes indeed I missed that one, fixed now!